src/components/MovieAdder.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useState, ChangeEvent } from "react";
+import AsyncSelect from "react-select/lib/Async";
+import { debounce } from "underscore";
+
+import { MoviePreference, User } from "../types";
+
+const promiseOptions = (inputValue: string) => {
+  const searchParams = new URLSearchParams({ search: inputValue });
+  return fetch("/search-movies?" + searchParams.toString())
+    .then(response => response.json())
+    .then(function(json: {
+      searchResults: Array<{ id: string; title: string }>;
+    }) {
+      return json.searchResults.map(option => ({
+        label: option.title,
+        value: option.id
+      }));
+    });
+};
+
+interface externalMovie {
```

Comment:
Could we just call it movie?

Reply:
Yeah...I was thinking it'd be nice to keep `external` if we wanted to add an internal one, but that difference probably isn't relevant on the front-end anyway